### PR TITLE
fix types for newest unified version

### PR DESCRIPTION
**Pull request by mashehu**
This should fix the following typescript errors (when using the latest version of unified in  `@astrojs/markdown-remark`):
```
@astrojs/markdown-remark:build:   Overload 1 of 3, '(preset?: Preset | null | undefined): Processor<Root, undefined, undefined, undefined, undefined>', gave the following error.
@astrojs/markdown-remark:build:     Type '(options?: void | Options | undefined) => void | Transformer<Root, Root>' has no properties in common with type 'Preset'.
@astrojs/markdown-remark:build:   Overload 2 of 3, '(list: PluggableList): Processor<Root, undefined, undefined, undefined, undefined>', gave the following error.
@astrojs/markdown-remark:build:     Argument of type '(options?: void | Options | undefined) => void | Transformer<Root, Root>' is not assignable to parameter of type 'PluggableList'.
@astrojs/markdown-remark:build:   Overload 3 of 3, '(plugin: Plugin<[], Root, Root>, ...parameters: [] | [boolean]): Processor<import("/Users/matho180/astro/node_modules/.pnpm/@types+mdast@4.0.2/node_modules/@types/mdast/index").Root, import("/Users/matho180/astro/node_modules/.pnpm/@types+mdast@3.0.14/node_modules/@types/mdast/index").Root, import("/Users/matho180/astro/node_modules/.pnpm/@types+mdast@3.0.14/node_modules/@types/mdast/index").Root, undefined, undefined>', gave the following error.
@astrojs/markdown-remark:build:     Argument of type '(options?: void | Options | undefined) => void | Transformer<Root, Root>' is not assignable to parameter of type 'Plugin<[], Root, Root>'.
@astrojs/markdown-remark:build:       Type 'void | Transformer<Root, Root>' is not assignable to type 'void | Transformer<Root, Root> | undefined'.
@astrojs/markdown-remark:build:         Type 'Transformer<Root, Root>' is not assignable to type 'void | Transformer<Root, Root> | undefined'.
```

### CODE REVIEW

**Comment by hahnbeelee** [comment]
Hi can this be merged? I'm also running into these issues

File: index.d.ts [hunk 1]
```diff
@@ -1,15 +1,20 @@
 /**
- * remark plugin to implement SmartyPants.
+ * Remark plugin to implement SmartyPants.
  *
```
Comment (silvenon) [inline index.d.ts]: I don't think this description does anything JSDoc-wise, so you can remove it.

File: index.d.ts [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
+ * Applies SmartyPants transformation to a Markdown Abstract Syntax Tree (MDAST).
+ *
+ * @param {SmartypantsOptions?} options - Configuration options for SmartyPants.
+ * @returns {Transformer<Root, Root>?} - A transformer for the MDAST.
+ */
 export default function remarkSmartypants(
-  options?: void | Options
-): void | Transformer<Root, Root>;
+  options?: SmartypantsOptions
+): Transformer<Root, Root> | undefined;
```
Comment (silvenon) [inline index.d.ts]: I think `void` needs to be included in this union as well 🤔 

https://github.com/unifiedjs/unified/blob/b69689bba52a918d87aa62f295ccffa8d9aa8ef8/lib/index.js#L112-L121

File: package.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
   "types": "./index.d.ts",
   "type": "module",
   "engines": {
-    "node": "^12.20.0 || ^14.13.1 || >=16.0.0"
```
Comment (silvenon) [inline package.json:14]: Which dependency does this v16 requirement come from?

**Review by silvenon — COMMENTED** [review]
Hey, thanks so much for efforst to keep this up-to-date, sorry about my late response. I pushed some updates to `main`, so please sync your branch and let me know if you need help.

File: index.d.ts [hunk 4]
```diff
@@ -1,15 +1,20 @@
 /**
- * remark plugin to implement SmartyPants.
+ * Remark plugin to implement SmartyPants.
```
Comment (mashehu) [inline index.d.ts:2]:
```suggestion
```

File: package.json (hunk 3 above)
Comment (mashehu) [reply package.json:14]: that's one of the major changes with the new version of all unified/remark packages, eg.: https://github.com/unifiedjs/unified/commit/baf80b20d604c920a0656b4eec1c17612689f5ba

**Comment by silvenon** [comment]
Can you resolve conflicts? Let me know if you need help.

**Comment by silvenon** [comment]
I'll continue this effort in a separate PR, probably by refactoring the source code itself to TypeScript and building type definitions that way. Seems like the safest way instead of maintaining them separately.

Btw, I seem to recall being able to use this plugin recently without type errors, though 🤔 maybe `@mdx-js/rollup` didn't have the latest `unified` dependency or whatever.

**Comment by silvenon** [comment]
Thanks for your contribution, @mashehu! I'll report back once I make a release. Whoever else is interested can follow #84.

**Pull request merged**